lib/modelManagers/readStatModel.js

```diff
@@ -0,0 +1,86 @@
+const { ReadStats, Articles } = require('../../database/models');
+
+/**
+ * Declares read stats model manager class
+ */
+class ReadStatsManager {
+  /**
+   *
+   * @param {object} newReadStat new read stat
+   * @returns {object} created readStat
+   */
+  static async createReadStat(newReadStat) {
+    const { articleId, userId } = newReadStat;
+    const createdReadStat = await ReadStats.findOrCreate({
+      where: {
+        articleId,
+        readerId: userId
+      }
+    });
+    return createdReadStat;
+  }
+
+  /**
+   *
+   * @param {object} readStatUpdate new read stat
+   * @returns {object} updated readStat
+   */
+  static async updateReadStat(readStatUpdate) {
+    const { articleId, userId, readStat } = readStatUpdate;
+    const updatedReadStat = await ReadStats.update(
+      { readStat },
+      {
+        where: {
+          articleId,
+          readerId: userId
+        },
+        returning: true,
+        plain: true
+      }
+    );
+    return updatedReadStat;
+  }
+
+  /**
+   *
+   * @param {object} searchOptions readerId, articleId and readStat
+   * @returns {object} readStat
+   */
+  static async getReadStat(searchOptions) {
+    const options = {
+      where: { ...searchOptions },
```

Comment:
```suggestion
      where: searchOptions,
```
Since you are not performing any mutation on `searchOptions` copying `searchOptions` to a new object is not necessary.